libsql-server/src/namespace/meta_store.rs

```diff
@@ -119,14 +131,17 @@ fn restore(db: &Connection) -> Result<HashMap<NamespaceName, Sender<Arc<Database
                 };
 
                 let config = match metadata::DatabaseConfig::decode(&v[..]) {
-                    Ok(c) => DatabaseConfig::from(&c),
+                    Ok(c) => Arc::new(DatabaseConfig::from(&c)),
                     Err(e) => {
                         tracing::warn!("unable to convert config: {}", e);
                         continue;
                     }
                 };
 
-                let (tx, _) = watch::channel(Arc::new(config));
+                // We don't store the version in the sqlitedb due to the session token
+                // changed each time we start the primary, this will cause the replica to
+                // handshake again and get the latest config.
+                let (tx, _) = watch::channel(InnerConfig { version: 0, config });
```

Comment:
this is only ok for now, because we keep the config for all namespaces in memory at all time. When we allow the number of namespaces to grow, we'll want to tie the config's lifetime to the namespace's lifetime in memory, which won't work anymore. To keep in mind for when we implement that.